File: web/src/types/board.ts

export type Board = {
    id: number;
    title: string;
    content: string;
    createdAt: string;
}

export type BoardSearchParams = {
    size: number,
    page: number,
}

export type BoardCreateRequest = {
    title: Board["title"];
    content?: Board["content"];
}

export type BoardUpdateRequest = {
    id: Board["id"];
    title: Board["title"];
    content?: Board["content"];
}

export type BoardUpdateFrom = {
    title: Board["title"];
    content?: Board["content"];
}

File: src/main/java/com/github/gun2/handsonspringshell/dto/Board.java
package com.github.gun2.handsonspringshell.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.LocalDateTime;

/**
 * 게시판 정보
 */
@Getter
@AllArgsConstructor
public class Board {
    private Long id;
    private String title;
    private String content;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public void update(String title, String content) {
        this.title = title;
        this.content = content;
        this.updatedAt = LocalDateTime.now();
    }
}
